dtool/Package.cmake

```diff
@@ -0,0 +1,703 @@
+      # this once that's gone
+      string(APPEND CMAKE_SHARED_LINKER_FLAGS " -undefined dynamic_lookup")
+
+    endif()
+
+  else()
+    # On every other platform, guessing is a bad idea - insist the user upgrade
+    # their CMake instead.
+    message(WARNING "For Python support on this platform, please use CMake >= 3.15!")
+    set(PYTHON_FOUND OFF)
+
+  endif()
+
+endif()
+
+package_option(PYTHON
+  DEFAULT ON
+  "Enables support for Python.  If INTERROGATE_PYTHON_INTERFACE
+is also enabled, Python bindings will be generated."
+  IMPORTED_AS Python::Module)
+
+# Also detect the optimal install paths:
+if(HAVE_PYTHON)
+  if(WIN32 AND NOT CYGWIN)
+    set(_LIB_DIR ".")
+    set(_ARCH_DIR ".")
+
+  elseif(PYTHON_EXECUTABLE)
+    execute_process(
+      COMMAND ${PYTHON_EXECUTABLE}
+        -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(False))"
+      OUTPUT_VARIABLE _LIB_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    execute_process(
+      COMMAND ${PYTHON_EXECUTABLE}
+        -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(True))"
+      OUTPUT_VARIABLE _ARCH_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  else()
+    set(_LIB_DIR "")
+    set(_ARCH_DIR "")
+
+  endif()
+
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE}
+      -c "from sysconfig import get_config_var as g; print((g('EXT_SUFFIX') or g('SO'))[:])"
+    OUTPUT_VARIABLE _EXT_SUFFIX
+    ERROR_QUIET
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  if(NOT _EXT_SUFFIX)
+    if(CYGWIN)
+      set(_EXT_SUFFIX ".dll")
+    elseif(WIN32)
+      set(_EXT_SUFFIX ".pyd")
+    else()
+      set(_EXT_SUFFIX ".so")
+    endif()
+  endif()
+
+  set(PYTHON_LIB_INSTALL_DIR "${_LIB_DIR}" CACHE STRING
+    "Path to the Python architecture-independent package directory.")
+
+  set(PYTHON_ARCH_INSTALL_DIR "${_ARCH_DIR}" CACHE STRING
+    "Path to the Python architecture-dependent package directory.")
+
+  set(PYTHON_EXTENSION_SUFFIX "${_EXT_SUFFIX}" CACHE STRING
+    "Suffix for Python binary extension modules.")
+
+endif()
+
+
+#
+# ------------ Data handling libraries ------------
+#
+
+# OpenSSL
+find_package(OpenSSL COMPONENTS SSL Crypto QUIET)
+
+package_option(OPENSSL
+  DEFAULT ON
+  "Enable OpenSSL support"
+  IMPORTED_AS OpenSSL::SSL OpenSSL::Crypto)
+
+option(REPORT_OPENSSL_ERRORS
+  "Define this true to include the OpenSSL code to report verbose
+error messages when they occur." ${IS_DEBUG_BUILD})
+
+if(REPORT_OPENSSL_ERRORS)
+  package_status(OPENSSL "OpenSSL" "with verbose error reporting")
+else()
+  package_status(OPENSSL "OpenSSL")
+endif()
+
+# zlib
+find_package(ZLIB QUIET)
+
+package_option(ZLIB
+  "Enables support for compression of Panda assets."
+  IMPORTED_AS ZLIB::ZLIB)
+
+package_status(ZLIB "zlib")
+
+
+#
+# ------------ Image formats ------------
+#
+
+# JPEG
+find_package(JPEG QUIET)
+
+package_option(JPEG DEFAULT ON "Enable support for loading .jpg images.")
```

Comment:
Since we support `stb_image` fallback for reading, we could remove the default-on status from PNG and JPEG—thoughts?

Reply:
Oh definitely. Default-on is meant to communicate "your user experience will be severely lacking if you don't have this; please consciously opt out of it if you don't have the package"  but if it's not necessary for the samples to work, we shouldn't be requiring it so heavily.

I never knew `stb_image` was a PNG/JPEG fallback - cool!

Reply:
Yeah; stb_image lacks support for writing images, and for some of the fancier features, but it does fine in the vast majority of use cases.